### PR TITLE
Change to folder path assumption for DSIronPython

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Update the folder path assumption for DSIronPython from testing as an individual package case. The new code should now work both with `DSIronPython.dll` exists in Dynamo root folder or together within the IronPythonExtension `bin` folder.

So what the package would look like?
image

image

TODO:
- [x] Confirm localization impact

Modules loaded Result:
image

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/IronPythonExtension/IronPythonExtension.cs [hunk 1]
```diff
@@ -37,15 +37,15 @@ public void Dispose()
...
             Assembly pythonEvaluatorLib = null;
             try
             {
-                pythonEvaluatorLib = Assembly.LoadFrom(Path.Combine(dynamoDir, PythonEvaluatorAssembly + ".dll"));
+                pythonEvaluatorLib = Assembly.LoadFrom(Path.Combine(targetDir, PythonEvaluatorAssembly + ".dll"));
             }
-            catch (Exception)
+            catch
             {
                 // Most likely the IronPython engine is excluded in this case
```
Comment (mjkkirschner) [inline src/IronPythonExtension/IronPythonExtension.cs:50]: @QilongTang would it be worth logging this? You can implement the ILogSource interface and log to the dynamo console quite easily from an extension if you wish.

Comment (mjkkirschner) [reply src/IronPythonExtension/IronPythonExtension.cs:50]: for example Docs browser does this:
https://github.com/DynamoDS/Dynamo/blob/5b086a92cab91dfead7ab6ec02862872fe8fcc3a/src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.cs#L43

Comment (QilongTang) [reply src/IronPythonExtension/IronPythonExtension.cs:63]: @mjkkirschner I have included the implementation but from my testing it does not work for extension?

**Comment by QilongTang** [comment]
@mjkkirschner Good to go?

**Pull request merged**